pkg/querier/queryrange/codec.go

```diff
@@ -270,19 +271,34 @@ func (Codec) DecodeRequest(_ context.Context, r *http.Request, forwardHeaders []
 			From:     from,
 			Through:  through,
 			Matchers: req.Query,
-		}, err
+		}, nil
 	default:
 		return nil, httpgrpc.Errorf(http.StatusBadRequest, fmt.Sprintf("unknown request path: %s", r.URL.Path))
 	}
 }
 
 func (Codec) EncodeRequest(ctx context.Context, r queryrangebase.Request) (*http.Request, error) {
 	header := make(http.Header)
-	queryTags := getQueryTags(ctx)
+
+	queryTags := httpreq.ExtractQueryTagsHeader(ctx)
 	if queryTags != "" {
 		header.Set(string(httpreq.QueryTagsHTTPHeader), queryTags)
 	}
 
+	actor := httpreq.ExtractHeader(ctx, httpreq.LokiActorPathHeader)
+	if actor != "" {
+		header.Set(httpreq.LokiActorPathHeader, actor)
+	}
```

Comment:
This is not strictly necessary.

Reply:
This sets the x-loki-actor-path HTTP header again when converting a LokiRequest back to a HTTP requests. That would ensure the header is also present in the httpgrcp.Request that is sent from the Frontend to the Scheduler. However, by adding the additional `QueuePath` field to the gRPC message, it's never read out in the scheduler.